### PR TITLE
[URP][Performance] Enable subset of perf tests in nightly jobs

### PR DESCRIPTION
**Pull request by sophiaaar**
### Purpose of this PR
This PR adds the changes needed to the ruamel in order for performance testing to work, as well as adding some of the performance jobs to the nightly.

* Takes build config into account in the Yamato jobs. Currently the existing graphics tests were set to Mono/.Net2 in the yml, because that's what they were already in master.
* Add performance jobs to nightly:
    * URP_Performance_BoatAttack on Android_OpenGLES3
    * URP_Performance_BoatAttack on Android_Vulkan
    * URP_Performance_BoatAttack on Win_DX11

---
### Testing status
Performance tests:
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/performance%252Fredo-metafiles/.yamato%252Fall-urp_performance_boatattack.yml%2523All_URP_Performance_BoatAttack_trunk/3841832/job/pipeline

---
### Comments to reviewers
This is PR 1 of the URP performance test initiative, which is why a lot of the metafiles have commented out sections - this is to show that these configs will be enabled later

### CODE REVIEW

File: .yamato/config/__shared.metafile [hunk 1]
```diff
@@ -58,9 +75,10 @@ platforms:
     apis:
       - name: Metal
         exclude_test_platforms:
-          - editmode
```
Comment (kivistikliis) [inline .yamato/config/__shared.metafile:61]: why do we allow editmode now or it's not intentional?

File: .yamato/config/__shared.metafile [hunk 2]
```diff
@@ -143,9 +153,8 @@ platforms:
     apis:
       - name: Metal
         exclude_test_platforms:
-        - editmode
```
Comment (kivistikliis) [inline .yamato/config/__shared.metafile:146]: wondering the same thing about editmode

File: .yamato/hdrp-linux-vulkan.yml [hunk 3]
```diff
@@ -322,7 +319,7 @@ Build_HDRP_Linux_Vulkan_Player_CUSTOM-REVISION:
...
       -  chmod +x TestProjects/HDRP_RuntimeTests/utr
       -  cd TestProjects/HDRP_RuntimeTests && sudo unity-downloader-cli --source-file ../../unity_revision.txt -c editor -c il2cpp  --wait --published-only
-      -  cd TestProjects/HDRP_RuntimeTests && DISPLAY=:0.0 ./utr --suite=playmode --platform=StandaloneLinux64 --testproject=. --editor-location=.Editor --artifacts_path=test-results --player-save-path=../../players --build-only --extra-editor-arg="-executemethod" --timeout=1200 --extra-editor-arg="CustomBuild.BuildLinuxVulkanLinear"
+      -  cd TestProjects/HDRP_RuntimeTests && DISPLAY=:0.0 ./utr --suite=playmode --platform=StandaloneLinux64 --testproject=. --extra-editor-arg="-playergraphicsapi=" --editor-location=.Editor --artifacts_path=test-results --player-save-path=../../players --build-only --extra-editor-arg="-scriptingbackend=il2cpp" --extra-editor-arg="-apicompatibilitylevel=NET_4_6" --extra-editor-arg="-colorspace=Linear" --extra-editor-arg="-executemethod" --timeout=1200 --extra-editor-arg="CustomBuild.BuildLinuxVulkanLinear"
```
Comment (kivistikliis) [inline .yamato/hdrp-linux-vulkan.yml:322]: should the flag --extra-editor-arg="-playergraphicsapi=" have something in the end?

File: .yamato/hdrp-osx-metal.yml [hunk 4]
```diff
@@ -30,8 +30,8 @@ HDRP_OSX_Metal_playmode_trunk:
             paths:
               -  "**/test-results/**"
               -  "TestProjects/HDRP_Tests/Logs/*.log"
-HDRP_OSX_Metal_playmode_XR_trunk:
```
Comment (kivistikliis) [inline .yamato/hdrp-osx-metal.yml:33]: similar to before, why are we excluding xr on metal?

File: .yamato/hdrp_hybrid-linux-vulkan.yml [hunk 5]
```diff
@@ -58,7 +57,7 @@ Build_HDRP_Hybrid_Linux_Vulkan_Player_trunk:
...
       -  chmod +x TestProjects/HDRP_HybridTests/utr
       -  cd TestProjects/HDRP_HybridTests && sudo unity-downloader-cli -u 5181ec54d595526303eb2714f480cbc3ee1911bd -c editor -c il2cpp  --wait --published-only
-      -  cd TestProjects/HDRP_HybridTests && DISPLAY=:0.0 ./utr --suite=playmode --platform=StandaloneLinux64 --testproject=. --editor-location=.Editor --artifacts_path=test-results --player-save-path=../../players --build-only --compilation-errors-as-warnings --extra-editor-arg="-executemethod" --timeout=1200 --extra-editor-arg="CustomBuild.BuildLinuxVulkanLinear"
+      -  cd TestProjects/HDRP_HybridTests && DISPLAY=:0.0 ./utr --suite=playmode --platform=StandaloneLinux64 --testproject=. --extra-editor-arg="-playergraphicsapi=" --editor-location=.Editor --artifacts_path=test-results --player-save-path=../../players --build-only --extra-editor-arg="-scriptingbackend=il2cpp" --extra-editor-arg="-apicompatibilitylevel=NET_4_6" --extra-editor-arg="-colorspace=Linear" --compilation-errors-as-warnings --extra-editor-arg="-executemethod" --timeout=1200 --extra-editor-arg="CustomBuild.BuildLinuxVulkanLinear"
```
Comment (kivistikliis) [inline .yamato/hdrp_hybrid-linux-vulkan.yml:60]: same here with the playergraphicsapi flag

File: .yamato/hdrp_hybrid-osx-metal.yml [hunk 6]
```diff
@@ -245,7 +160,7 @@ HDRP_Hybrid_OSX_Metal_Standalone_CUSTOM-REVISION:
...
+                 ssh -i ~/.ssh/id_rsa_macmini -o "StrictHostKeyChecking=no" bokken@$BOKKEN_DEVICE_IP "export UPM_REGISTRY=https://artifactory-slo.bf.unity3d.com/artifactory/api/npm/upm-candidates; echo \$UPM_REGISTRY; cd ~/Graphics/TestProjects/HDRP_HybridTests && ~/Graphics/TestProjects/HDRP_HybridTests/utr --suite=playmode --platform=StandaloneOSX --artifacts_path=/Users/bokken/Graphics/TestProjects/HDRP_HybridTests/test-results --testproject=/Users/bokken/Graphics/TestProjects/HDRP_HybridTests --editor-location=/Users/bokken/.Editor --extra-editor-arg="-scriptingbackend=mono" --extra-editor-arg="-apicompatibilitylevel=NET_2_0" --extra-editor-arg="-colorspace=Linear"--reruncount=2 --compilation-errors-as-warnings --timeout=2400"
```
Comment (kivistikliis) [inline .yamato/hdrp_hybrid-osx-metal.yml:163]: there is a missing space between `--extra-editor-arg="-colorspace=Linear"--reruncount=2`, probably counts for all metals?

File: .yamato/config/universal.metafile [hunk 7]
```diff
@@ -11,6 +11,14 @@ test_platforms:
       Win: 2000
   - type: playmode
   - type: editmode
+  - type: playmode
```
Comment (kivistikliis) [inline .yamato/config/universal.metafile:14]: i think this testplatform can be removed? i think it got here accidentally from my example branch

File: .yamato/universal-android-opengles3.yml [hunk 8]
```diff
@@ -54,8 +53,108 @@ Build_Universal_Android_OpenGLES3_Player_trunk:
         players:
             paths:
               -  "players/**"
-Universal_Android_OpenGLES3_Standalone_2020.2:
-    name: Universal on Android_OpenGLES3_Standalone on version 2020.2
+Universal_Android_OpenGLES3_playmode_mono_2_Linear_trunk:
```
Comment (kivistikliis) [inline .yamato/universal-android-opengles3.yml:56]: do we want playmodes for android?

File: .yamato/config/__shared.metafile (hunk 1 above)
Comment (sophiaaar) [reply .yamato/config/__shared.metafile:61]: because the performance build tests are editmode tests

File: .yamato/config/__shared.metafile (hunk 2 above)
Comment (sophiaaar) [reply .yamato/config/__shared.metafile:146]: because the performance build tests are editmode tests

File: .yamato/hdrp-linux-vulkan.yml (hunk 3 above)
Comment (sophiaaar) [reply .yamato/hdrp-linux-vulkan.yml:322]: good catch, let me check this

File: .yamato/hdrp-osx-metal.yml (hunk 4 above)
Comment (sophiaaar) [reply .yamato/hdrp-osx-metal.yml:33]: oh i thought this was a change that happened in trunk 🤔

File: .yamato/config/__shared.metafile [hunk 9]
```diff
@@ -115,15 +133,7 @@ platforms:
     os: android
     apis:
       - name: OpenGLES3
-        exclude_test_platforms:
```
Comment (kivistikliis) [inline .yamato/config/__shared.metafile:118]: should these be added back, or we want editmodes/playmodes on mobiles?

File: .yamato/config/__shared.metafile (hunk 2 above)
Comment (kivistikliis) [inline .yamato/config/__shared.metafile:146]: here as well

File: .yamato/hdrp-osx-metal.yml (hunk 4 above)
Comment (sophiaaar) [reply .yamato/hdrp-osx-metal.yml:33]: oh i guess not, i will undo this

File: .yamato/hdrp_hybrid-linux-vulkan.yml (hunk 5 above)
Comment (sophiaaar) [reply .yamato/hdrp_hybrid-linux-vulkan.yml:60]: fixed (havent pushed yet) - it was an issue that happened with the merge and linux

File: .yamato/hdrp_hybrid-osx-metal.yml (hunk 6 above)
Comment (sophiaaar) [reply .yamato/hdrp_hybrid-osx-metal.yml:163]: ah! there was a missing comma in utr_utils - fixed!

File: .yamato/config/__shared.metafile (hunk 9 above)
Comment (sophiaaar) [reply .yamato/config/__shared.metafile:118]: see prev comments

File: .yamato/config/__shared.metafile (hunk 1 above)
Comment (kivistikliis) [reply .yamato/config/__shared.metafile:61]: true, but we call these performance build tests with some other name, right? i think the matching is done by testplatform name, i.e. 'if testplatform[name] == editmode then skip", so removing this line means all normal editmode jobs also get created (before we threw exception in the _cmd functions, which made it impossible to use editmode for performance, but we removed this i think), which is why I think we should add this back (the performance jobs should not be affected as long as they are not named editmode)

File: .yamato/config/__shared.metafile (hunk 2 above)
Comment (sophiaaar) [reply .yamato/config/__shared.metafile:146]: see prev comments

File: .yamato/config/__shared.metafile (hunk 1 above)
Comment (sophiaaar) [reply .yamato/config/__shared.metafile:61]: i see your point - and to be fair, im not even enabling the build jobs in this pr, so i can add this bak in anyway and we can figure it out after

File: .yamato/all-universal.yml [hunk 10]
```diff
@@ -6,17 +6,15 @@
...
-      -  path: .yamato/universal-win-dx11.yml#Universal_Win_DX11_Standalone_trunk
+      -  path: .yamato/universal-win-dx11.yml#Universal_Win_DX11_Standalone_mono_2_Linear_trunk
          rerun: on-new-revision
-      -  path: .yamato/universal-win-dx11.yml#Universal_Win_DX11_editmode_trunk
+      -  path: .yamato/universal-win-dx11.yml#Universal_Win_DX11_editmode_mono_2_Linear_trunk
          rerun: on-new-revision
-      -  path: .yamato/universal-win-dx11.yml#Universal_Win_DX11_playmode_trunk
+      -  path: .yamato/universal-win-dx11.yml#Universal_Win_DX11_playmode_mono_2_Linear_trunk
          rerun: on-new-revision
-      -  path: .yamato/universal-win-dx11.yml#Universal_Win_DX11_playmode_XR_trunk
```
Comment (theo-pnv) [inline .yamato/all-universal.yml:15]: I think `Universal_Win_DX11_playmode_XR_trunk` is missing, is it wanted? Same for 2020.2 and custom-revision

File: .yamato/config/_abv.metafile [hunk 11]
```diff
@@ -5,84 +5,119 @@ abv: # specify the editor versions for which to create the expression trigger fo
...
     - project: Universal
       platform: Win
       api: DX12
+      build_config: mono_2
+      color_space: Linear
       test_platforms:
         - Standalone
         - editmode
         - playmode
-        # - playmode_XR # uncomment once bug fixed on XR side
```
Comment (theo-pnv) [inline .yamato/config/_abv.metafile:45]: Related to a previous comment. so I guess this is intentional. Is the XR issue related to perf testing? iirc it works correctly currently.

File: .yamato/hdrp-win-dx12.yml [hunk 12]
```diff
@@ -72,14 +72,14 @@ HDRP_Win_DX12_editmode_trunk:
       -  curl -s https://artifactory.internal.unity3d.com/core-automation/tools/utr-standalone/utr.bat --output TestProjects/HDRP_Tests/utr.bat
       -  pip install unity-downloader-cli --index-url https://artifactory.prd.it.unity3d.com/artifactory/api/pypi/pypi/simple --upgrade
       -  cd TestProjects/HDRP_Tests && unity-downloader-cli -u 5181ec54d595526303eb2714f480cbc3ee1911bd -c editor -c il2cpp  --wait --published-only
-      -  cd TestProjects/HDRP_Tests && utr --suite=editor --platform=editmode --testproject=. --editor-location=.Editor --artifacts_path=test-results --reruncount=2 --extra-editor-arg="-force-d3d12"
+      -  cd TestProjects/HDRP_Tests && utr --suite=editor --platform=editmode --testproject=. --editor-location=.Editor --artifacts_path=test-results --scripting-backend=il2cpp --extra-editor-arg="-colorspace=Linear" --reruncount=2 --extra-editor-arg="-force-d3d12"
```
Comment (theo-pnv) [inline .yamato/hdrp-win-dx12.yml:75]: `--scripting-backend=il2cpp` but the name of this test is `mono_2`

File: .yamato/shadergraph-linux-openglcore.yml [hunk 13]
```diff
@@ -37,14 +37,14 @@ ShaderGraph_Linux_OpenGLCore_editmode_trunk:
       -  curl -s https://artifactory.internal.unity3d.com/core-automation/tools/utr-standalone/utr --output TestProjects/ShaderGraph/utr
       -  chmod +x TestProjects/ShaderGraph/utr
       -  cd TestProjects/ShaderGraph && sudo unity-downloader-cli -u 5181ec54d595526303eb2714f480cbc3ee1911bd -c editor -c il2cpp  --wait --published-only
-      -  cd TestProjects/ShaderGraph && DISPLAY=:0.0 ./utr --suite=editor --platform=editmode --testproject=. --editor-location=.Editor --artifacts_path=test-results --reruncount=2 --extra-editor-arg="-force-glcore"
+      -  cd TestProjects/ShaderGraph && DISPLAY=:0.0 ./utr --suite=editor --platform=editmode --testproject=. --editor-location=.Editor --artifacts_path=test-results --scripting-backend=il2cpp --extra-editor-arg="-colorspace=Linear" --reruncount=2 --extra-editor-arg="-force-glcore"
```
Comment (theo-pnv) [inline .yamato/shadergraph-linux-openglcore.yml:40]: Same about il2cpp and mono (it happens in many places/different configs)

File: .yamato/hdrp-win-dx12.yml (hunk 12 above)
Comment (sophiaaar) [reply .yamato/hdrp-win-dx12.yml:75]: ah! good catch, thank you

Comment (sophiaaar) [reply .yamato/hdrp-win-dx12.yml:75]: weird, im not seeing this locally?

Comment (sophiaaar) [reply .yamato/hdrp-win-dx12.yml:75]: ah, found it

Comment (sophiaaar) [reply .yamato/hdrp-win-dx12.yml]: fixed

File: .yamato/shadergraph-linux-openglcore.yml (hunk 13 above)
Comment (sophiaaar) [reply .yamato/shadergraph-linux-openglcore.yml]: fixed

File: .yamato/config/_abv.metafile (hunk 11 above)
Comment (sophiaaar) [reply .yamato/config/_abv.metafile:45]: oh its commented out on master though,

File: .yamato/all-universal.yml (hunk 10 above)
Comment (sophiaaar) [reply .yamato/all-universal.yml:15]: accidental i think, checking

**Review by theo-pnv — APPROVED** [review]

**Comment by sophiaaar** [comment]
canelling the non-essential jobs (they have passed on previous revisions)

**Pull request merged**